Commentservice/src/main/java/com/nagarro/Commentservice/services/impl/CommentServiceImpl.java

```diff
@@ -12,47 +13,82 @@
 import com.nagarro.Commentservice.Exceptions.CommentNotFoundException;
 import com.nagarro.Commentservice.Exceptions.InvalidCommentException;
+import com.nagarro.Commentservice.Exceptions.InvalidNotificationRequestException;
 import com.nagarro.Commentservice.Exceptions.InvalidRequestException;
 import com.nagarro.Commentservice.models.Comment;
 import com.nagarro.Commentservice.models.User;
 import com.nagarro.Commentservice.repository.CommentRepository;
 import com.nagarro.Commentservice.services.CommentService;
 import com.nagarro.Commentservice.services.JWTService;
 
+import reactor.core.publisher.Mono;
+
 @Service
 public class CommentServiceImpl implements CommentService {
 
 	private JWTService jwtService;
 	private CommentRepository commentRepository;
 	private Validator validator;
 	private final WebClient webClient;
+	private final WebClient notificationWebClient;
 	
 	public CommentServiceImpl(JWTService jwtService, CommentRepository commentRepository, Validator validator, WebClient.Builder webClientBuilder) {
 		this.jwtService = jwtService;
 		this.commentRepository = commentRepository;
 		this.validator=validator;
 		this.webClient = webClientBuilder.baseUrl("http://localhost:8181").build();
+		this.notificationWebClient = webClientBuilder.baseUrl("http://localhost:8084").build();
 	}
 	
 	@Override
-	public Comment addComment(CommentDTO commentDTO, String token) throws InvalidCommentException {
+	public Comment addComment(CommentDTO commentDTO, String token) throws InvalidCommentException, InvalidNotificationRequestException {
 		validateComment(commentDTO);
 		String id = (String) jwtService.decodeJWT(token).get("jti");
 		Comment comment = new Comment();
 		comment.setPostId(commentDTO.getPostId());
 		comment.setPostAuthorId(commentDTO.getPostAuthorId());
 		comment.setCommentAuthorId(id);
 		comment.setContent(commentDTO.getContent());
+		User user = getUser(token, id);
+		sendNotification(user.getName(), comment).subscribe();
 		return this.commentRepository.save(comment);
 	}
 	
+	private Mono<Void> sendNotification(String username, Comment comment) throws InvalidNotificationRequestException {
+        NotificationRequestDTO notification = new NotificationRequestDTO();
+        notification.setContent(username+ " added a comment on your post!");
+        notification.setSender(username);
+        notification.setReceiver(comment.getPostAuthorId());
+        notification.setGroupNotification(false);
+        notification.setTimestamp(LocalDateTime.now());
+        System.out.println(notification);
+        validateNotification(notification);
+        
+        return notificationWebClient.post()
+                .uri("/sendNotification")
+                .body(BodyInserters.fromValue(notification))
+                .retrieve()
+                .toBodilessEntity()
+                .then();
+    }
+	 
+	private void validateNotification(NotificationRequestDTO notification) throws InvalidNotificationRequestException {
```

Comment:
This is redundant because Notification service already does validations. Instead of this handle WebClientException when making the request